go-controller/pkg/cni/helper_linux.go

```diff
@@ -362,92 +376,108 @@ func ConfigureOVS(ctx context.Context, namespace, podName, hostIfaceName string,
 	// Find and remove any existing OVS port with this iface-id. Pods can
 	// have multiple sandboxes if some are waiting for garbage collection,
 	// but only the latest one should have the iface-id set.
-	names, _ := ovsFind("Interface", "name", "external-ids:iface-id="+ifaceID)
-	for _, name := range names {
-		if name == hostIfaceName {
+	p := func(iface *vswitchdb.Interface) bool {
+		if iface.Name == hostIfaceName {
 			// this may be result of restarting ovnkube-node, and it is trying to add the same VF representor to
 			// br-int for the same pod; do not delete port in this case.
-			continue
-		}
-		if out, err := ovsExec("--with-iface", "del-port", "br-int", name); err != nil {
-			klog.Warningf("Failed to delete stale OVS port %q with iface-id %q from br-int: %v\n %q",
-				name, ifaceID, err, out)
+			return false
 		}
+		foundID, ok := iface.ExternalIDs["iface-id"]
+		return ok && foundID == ifaceID
+	}
+	if err := libovsdbops.DeleteInterfacesWithPredicate(vsClient, p); err != nil {
```

Comment:
nit: you define trhe predicate before the function call where just below you define the predicate in the function call. Why not set this up the same both times?